subsys/net/lib/sockets/sockets.c

```diff
@@ -56,11 +58,27 @@ static void zsock_flush_queue(struct net_context *ctx)
 	k_fifo_cancel_wait(&ctx->recv_q);
 }
 
+static inline struct net_context *sock_to_net_ctx(int sock)
+{
+	return z_get_fd_obj(sock, &sock_fd_op_vtable, ENOTSOCK);
+}
+
 int _impl_zsock_socket(int family, int type, int proto)
 {
+	int fd = z_reserve_fd();
 	struct net_context *ctx;
+	int res;
 
-	SET_ERRNO(net_context_get(family, type, proto, &ctx));
+	if (fd < 0) {
```

Comment:
is z_free_fd(fd) needed here?

Reply:
OK

Reply:
> we follow ugly, unnatural rules of defining variables not where they are used, but possibly hundreds of lines away. I wonder, what MISRA thinks of this errorprone way.

Disagree here, it is very convenient if variables are found only in one place instead of scattered all over the function.